types/nightwatch/index.d.ts

```diff
@@ -5736,24 +5771,32 @@ export interface WebDriverProtocolNavigation {
      *     });
      *   }
      * }
+     *
+     * @see https://nightwatchjs.org/api/url.html
      */
-    url(url?: string, callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void): this;
-    url(callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void): this;
+    url(callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void): Awaitable<this, string>;
+    url(url?: string, callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void): Awaitable<this, null>;
```

Comment:
I think what I meant here was if we set url as `undefined` (not as `null`), it will return the current url (string) and not `null`. So, we don't need to make it `url: string | null` here, which is anyways not right because setting url to string should return `null` but it will return `string | null`.

We can just keep the second overload to:
```ts
url(url: string, callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void): Awaitable<this, null>;
```